tools/downloader/src/open_model_zoo/model_tools/_configuration.py

```diff
@@ -359,9 +360,23 @@ def deserialize(cls, model, name, subdirectory):
             task_type = validate_string_enum('"task_type"', model['task_type'],
                 _common.KNOWN_TASK_TYPES)
 
+            composite_model_name = model.get('composite_model_name', None)
+            if composite_model_name is not None:
+                composite_model_name = validate_string('"composite_model_name"', composite_model_name)
+
             return cls(name, subdirectory, files, postprocessing, mo_args, framework,
                 description, license_url, precisions, quantization_output_precisions,
-                task_type, conversion_to_onnx_args)
+                task_type, conversion_to_onnx_args, composite_model_name)
+
+def check_composite_model_dir(model_dir):
+    if list(model_dir.glob('*/*/*')):
+        raise DeserializationError('{} directory should not contain any other folders'.format(model_dir))
```

Comment:
This check seems too broad. The only thing that really matters is that there should not be any `model.yml` files in any subdirectories that are not direct children of the composite model directory (as well as in the composite model directory itself).